edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/requests/ModuleLogsRequestHandler.cs

```diff
@@ -58,25 +61,26 @@ protected override async Task<Option<IEnumerable<ModuleLogsResponse>>> HandleReq
 
                     if (l.logOptions.ContentEncoding == LogsContentEncoding.Gzip)
                     {
-                        if (moduleLogs.Length > MaxPayloadSize)
-                        {
-                            return new ModuleLogsResponse(l.id, Events.LargePayload(moduleLogs.Length, l.logOptions));
-                        }
-
+                        messageSize += moduleLogs.Length;
```

Comment:
There are potentially multiple module log requests being processed simultaneously here...is there a possibility that two threads will try to update `messageSize` at the same time and corrupt its value?

Reply:
If so, we could sum up message sizes after `Task.WhenAll` below, rather than using a sync primitive...

Reply:
I thought about using Interlocked.Add, but then I read that int read/write was atomic. Doing another quick on the internet, it looks like I misinterpreted that and read-modify write isn't atomic. Ill replace it.